### PR TITLE
[DO-NOT-MERGE-YET][Dependencies] CUDA compatibility issues in linux

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*
CUDA compatibility issues in linux - fixing torch to < 1.8.x
This fixeds `CUDNN_STATUS_NOT_INITIALIZE` errors during inference.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jwmueller — APPROVED** [review]

File: tabular/setup.py [hunk 1]
```diff
@@ -44,7 +44,8 @@
         'xgboost>=1.3.2,<1.4',
     ],
     'fastai': [
-        'torch>=1.0,<2.0',
+        'torch>=1.7.*,<1.8.0',
```
Comment (Innixma) [inline tabular/setup.py:47]: We should support torch 1.6 as well

Comment (gradientsky) [reply tabular/setup.py]: updated

**Review by Innixma — APPROVED** [review]

**Comment by Innixma** [comment]
When does this error occur? I just tested on linux with torch 1.8 (no GPU):

```
>>> predictions = predictor.predict(test_data)
>>> leaderboard = predictor.leaderboard(test_data)
                 model  score_test  score_val  pred_time_test  pred_time_val   fit_time  pred_time_test_marginal  pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  fit_order
0      NeuralNetFastAI     0.85976     0.8648        0.583807       0.179888  53.181877                 0.583807                0.179888          53.181877            1       True          1
1  WeightedEnsemble_L2     0.85976     0.8648        0.587145       0.186265  53.185163                 0.003338                0.006376           0.003286            2       True          2
```

**Comment by gradientsky** [comment]
> When does this error occur? I just tested on linux with torch 1.8 (no GPU):
It works on CPU instances. The issue happens when GPU is available and only during inference time.

**Comment by gradientsky** [comment]
Tracking:
https://github.com/pytorch/pytorch/issues/53336
https://github.com/pytorch/pytorch/issues/54047

**Comment by szha** [comment]
Job PR-1023-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1023/2/index.html

**Pull request closed without merging**